### PR TITLE
Assets library base, improvements

### PR DESCRIPTION
**Pull request by imgrby**
.. and JS/.less optimization

### CODE REVIEW

File: browser/scripts/checkbrowser.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+checkBrowser = function() {
+	var agent = navigator.userAgent;
+	if ((/Chrome/i.test(agent)) || (/Firefox/i.test(agent))) {
+		// Good boy browser
+	}   
+    else if (/Android|webOS|CriOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini|Mobile|mobile/i.test(agent)) {
```
Comment (kschzt) [inline browser/scripts/checkbrowser.js]: you can use E2.util.isMobile()

File: browser/scripts/checkbrowser.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+checkBrowser = function() {
```
Comment (kschzt) [inline browser/scripts/checkbrowser.js]: instead of creating a global function, please use eg. E2.app.ui namespace

File: browser/scripts/application.js [hunk 3]
```diff
@@ -52,36 +52,24 @@ function Application() {
...
-
 	this.graphStore = new GraphStore()
 	this.peopleStore = new PeopleStore()
-
 	this.peopleManager = new PeopleManager(this.peopleStore, $('#peopleTab'))
 
 	// Make the UI visible now that we know that we can execute JS
 	$('.nodisplay').removeClass('nodisplay');
 	
-	if (this.viewMode==='editor') {
```
Comment (kschzt) [inline browser/scripts/application.js]: why did you remove the viewMode?

Comment (imgrby) [inline browser/scripts/application.js]: misunderstanding with design. first it was 'editor/camera' switch button. than, in later design, instead of these two buttons, there was new 'code' button. now it should be 'code/patches', so viewMode will be back.

File: browser/scripts/checkbrowser.js (hunk 2 above)
Comment (imgrby) [inline browser/scripts/checkbrowser.js]: so if something from E2.app fails in unsupported browser - user will never see the message. that's why it should be called before E2 initialization

File: browser/scripts/checkbrowser.js (hunk 1 above)
Comment (imgrby) [inline browser/scripts/checkbrowser.js]: well, as long as util.js loaded vefore application.js, I think whole thing can be moved there.

File: browser/scripts/checkbrowser.js (hunk 2 above)
Comment (imgrby) [inline browser/scripts/checkbrowser.js]: function moved to E2.util.

File: package.json [hunk 4]
```diff
@@ -47,8 +47,8 @@
     "mime": "~1.2.11",
     "minimist": "0.0.8",
     "moment": "^2.9.0",
-    "mongodb": "~1.4.35",
-    "mongoose": "~3.8.26",
+    "mongodb": "2.0.42",
```
Comment (kschzt) [inline package.json]: please revert these changes to the state in `develop` :)

**Comment by kschzt** [comment]
except for the package.json, lgtm
